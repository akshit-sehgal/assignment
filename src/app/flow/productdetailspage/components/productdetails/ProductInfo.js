import React from 'react';
import PropTypes from 'prop-types';
import './productinfo.scss';
import { DETAILS_LABEL } from './constants';

const StarRating = ({ count }) => {
    const stars = [];
    for (let i = 0; i < count; i++) {
        stars.push(
            <i className="material-icons" key={i}>
                star
            </i>
        )
    }

    return stars;
}
StarRating.defaultProps = {
    count: 0
}
StarRating.propTypes = {
    count: PropTypes.number.isRequired
}


const ProductInfo = (props) => {
    const { name, rating, price, description } = props;
    return (
        <div className="pd__info-container">
            <table className="pf__info-table">
                <tbody>
                    <tr className="pf__table-row">
                        <td>{DETAILS_LABEL.NAME}</td>
                        <td>{name}</td>
                    </tr>
                    <tr className="pf__table-row">
                        <td>{DETAILS_LABEL.RATING}</td>
                        <td className="pd__rating">
                            <StarRating count={rating} />
                        </td>
                    </tr>
                    <tr className="pf__table-row">
                        <td>{DETAILS_LABEL.PRICE}</td>
                        <td>{`${price} /-`}</td>
                    </tr>
                    <tr className="pf__table-row">
                        <td>{DETAILS_LABEL.DESCRIPTION}</td>
                        <td className="pd__desc">{description}</td>
                    </tr>
                </tbody>
            </table>
        </div>
    );
}
ProductInfo.propTypes = {
    rating: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
}
export default ProductInfo;