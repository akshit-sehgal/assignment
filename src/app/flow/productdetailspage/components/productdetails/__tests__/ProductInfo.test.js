import React from 'react';
import { shallow } from 'enzyme';
import ProductInfo from '../ProductInfo';
import { mockProductDetails } from '../__mocks__/productDetails.mock';

describe('Testing <ProductImage />', () => {
    let props;
    beforeEach(() => {
        props = {
            rating: mockProductDetails.rating,
            name: mockProductDetails.name,
            price: mockProductDetails.price,
            description: mockProductDetails.description
        }
    })
    it('should render the default view', () => {
        const wrapper = shallow(<ProductInfo {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.pf__table-row').length).toEqual(4);
        expect(wrapper.find('.pd__rating').length).toEqual(1);
        expect(wrapper.find('.pd__desc').length).toEqual(1);
        const StarRattingWrapper = wrapper.find('StarRating').dive();
        expect(StarRattingWrapper.find('.material-icons').length).toEqual(props.rating);
        expect(StarRattingWrapper).toMatchSnapshot();
    });

});