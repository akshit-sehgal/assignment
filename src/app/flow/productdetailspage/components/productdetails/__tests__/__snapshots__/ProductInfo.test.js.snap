// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ProductImage /> should render the default view 1`] = `
<div
  className="pd__info-container"
>
  <table
    className="pf__info-table"
  >
    <tbody>
      <tr
        className="pf__table-row"
      >
        <td>
          Product Name
        </td>
        <td>
          Product 1
        </td>
      </tr>
      <tr
        className="pf__table-row"
      >
        <td>
          Rating
        </td>
        <td
          className="pd__rating"
        >
          <StarRating
            count={4}
          />
        </td>
      </tr>
      <tr
        className="pf__table-row"
      >
        <td>
          Price
        </td>
        <td>
          Rs 2000 /-
        </td>
      </tr>
      <tr
        className="pf__table-row"
      >
        <td>
          Description
        </td>
        <td
          className="pd__desc"
        >
          Product Description
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Testing <ProductImage /> should render the default view 2`] = `
Array [
  <i
    className="material-icons"
    key="0"
  >
    star
  </i>,
  <i
    className="material-icons"
    key="1"
  >
    star
  </i>,
  <i
    className="material-icons"
    key="2"
  >
    star
  </i>,
  <i
    className="material-icons"
    key="3"
  >
    star
  </i>,
]
`;
